frontend/src/metabase/dashboard/hooks/use-dashboard-refresh-period.ts

```diff
@@ -16,33 +17,41 @@ export const useDashboardRefreshPeriod = ({
   const elapsedHook = useRef<((elapsed: number | null) => void) | null>(null);
   const elapsed = useRef<number | null>(0);
 
-  const setRefreshElapsedHook = (hook: (elapsed: number | null) => void) => {
-    elapsedHook.current = hook;
-  };
+  const setRefreshElapsedHook = useCallback(
+    (hook: (elapsed: number | null) => void) => {
+      elapsedHook.current = hook;
+    },
```

Comment:
Do we need the useCallback here? Curious since there's no dependencies

Reply:
every time you run this hook, new function will be created and new reference will be returned, so yes

Reply:
see the screenshot at `Before` section in the description, it shows that this function had a new reference

Reply:
Ah yes of course. Thanks 😄